Tests/FluentAssertions.Equivalency.Specs/CyclicReferencesSpecs.cs

```diff
@@ -310,79 +229,6 @@ IEnumerator IEnumerable.GetEnumerator()
         }
     }
 
-    internal class LogbookEntryProjection
-    {
-        public virtual LogbookCode Logbook { get; set; }
-
-        public virtual ICollection<LogbookRelation> LogbookRelations { get; set; }
-    }
-
-    internal class LogbookRelation
-    {
-        public virtual LogbookCode Logbook { get; set; }
-    }
-
-    internal class LogbookCode
-    {
-        public LogbookCode(string key)
-        {
-            Key = key;
-        }
-
-        public string Key { get; protected set; }
-    }
-
-    [Fact]
-    public void When_the_root_object_is_referenced_from_a_nested_object_it_should_treat_it_as_a_cyclic_reference()
```

Comment:
Is this covered elsewhere?
It looks related to `When_the_root_object_is_referenced_from_an_object_in_a_nested_collection_it_should_treat_it_as_a_cyclic_reference` which is the only other test that references `MyCompany`, `MyUser` and `MyCompanyLogo`.

Reply:
I couldn't think of a scenario where this test is doing something else than `Cyclic_references_can_be_ignored_in_equivalency_assertions` and the code coverage confirmed that.

Reply:
There are limits to what information code coverage can provide 🙃 

`When_the_root_object_is_referenced_from_a_nested_object_it_should_treat_it_as_a_cyclic_reference` was introduced in 02e71b30159e34b0484dac40589633fe49256466 and is one of three tests to exercise `IgnoringCyclicReferences()`

`When_asserting_equivilence_on_objects_needing_high_recursion_depth_and_disabling_recursion_depth_limit_it_should_recurse_to_completion` was introduced in 36740ca412815e2c90a53632c88af00969fdb07e and is one of three tests to exercise `AllowingInfiniteRecursion()`

Reply:
There's another `When_the_root_object_is_referenced_from_a_nested_object_it_should_treat_it_as_a_cyclic_reference` as well, which is doing the same thing.

Reply:
Just to be on the safe side, I've restored that test, but renamed it to `Can_detect_cyclic_references_in_nested_objects_referring_to_the_root`